File: proxy/reconf.pl
#!/usr/bin/perl

use strict;

my $filename=$ARGV[0] ? $ARGV[0] : "config.yml";

open F1, '<', $filename or die $filename . " open failed.\n";
my @lines = <F1>;
close F1;
my $str = join "", @lines;
# allow lan
$str =~ s/allow-lan: false/allow-lan: true/mg;
# use auto group instead
$str =~ s/name: Proxy\n  type: select/name: Proxy\n  type: url-test\n  url: http:\/\/www\.gstatic\.com\/generate_204\n  interval: 300/smg;    # about perl multi-line match, refer to zebos/script/rpc*_mod.pl

open F2, '>', $filename or die $filename . " open failed.\n";
print F2 $str;
close F2;

File: proxy/reconf.lua
src=arg[1] or "config.yaml"
dst=arg[2] or src
-- read
f = io.open(src, "r")
text = f:read("*a")
f:close()

-- proc
text = string.gsub(text, "\r", "")
-- remove game server, for game server's access to google is not very stable
text = string.gsub(text, "[^\n]+游戏[^\n]+\n", "")
-- allow lan
text = string.gsub(text, "allow-lan: false", "allow-lan: true")
if string.find(text, "url%-test") == nil then
    text = string.gsub(text, "节点选择\n    type: select", "节点选择\n    type: url-test\n    url: http://www.gstatic.com/generate_204\n    interval: 300")
end

-- write
f = io.open(dst, "w")
text = f:write(text)
f:close()
